### ISSUE TITLE
Empty library after upgrade from Kodi 19.4 to Kodi 20

### ISSUE THREAD

**Issue by Raptor039** [xbmc/xbmc#22510]
### Describe the bug

Libraries are empty after upgrade from Kodi 19.4 to Kodi 20.
Libraries are stored on a shared mysql (mariadb / debian 11 up to date) linked with a second Android TV.

## Expected Behavior

Recover my library wich was working well when my TV ran Kodi 19.4

## Actual Behavior

Libraries are empty. I'm unable to refresh them. No error seems to appear on the interface.

## Possible Fix

I don't know? Maybe an issue with the mysql migration script?

### To Reproduce

1. Update from Kodi 19.4 to Kodi 20
2. MySQL database is upgraded with no error apparently
3. Libraries are empty at Kodi 20 first launch

### Debuglog

The debuglog can be found here:

https://pastebin.com/jyhmZuzT

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android TV 9 (Kodi) + mariadb server 10.5.15 on Rasperry Pi Debian 11 up to date (mysql_upgrade already ok)
 - Kodi version: 20

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by dirkdirk** [comment]
Same experience here on NVidia Shield Pro running Android TV 9.1.1 with external SSD drive.

**Comment by Raptor039** [comment]
> Same experience here on NVidia Shield Pro running Android TV 9.1.1 with external SSD drive.
> 
> Found the solution ...
> 
> On Android TV go to Settings -> Apps - Kodi -> Permissions -> Files and Media and select Allow all the time.

I tried that yesterday. It didn't do the trick for me.

**Comment by dirkdirk** [comment]
Try going to the Android settings, find the drives section, there's a toggle button for "Automatically scan for media files" or something like that.  Make sure that is active.

Another thing to try is, install VLC player (or some other media player) and check it that app can see media files.  That way we can at least narrow it down to either an OS issue or an app issue.

Let me know if you can't find it and I'll post more exact instructions when I have my tv running.

**Comment by Raptor039** [comment]
Hi @dirkdirk,

I can browse the smb share (my NAS) through kodi without any problem by manualy going to files->sources and get into my Videos sources which were configured as movie or TV show source for the kodi library :

![Screenshot_20230118-223708.png](https://user-images.githubusercontent.com/16542820/213300464-6ad0eea6-3c8a-4710-a214-674e3bed0f19.png)

I tried to set up a fresh kodi 20 on a linux VM and import my advancedsettings.xml to migrate the MySQL database but the result is exactly the same as on Android.

However, all is working great if I set up a new kodi 19. 

I don't believe this problem is coming from OS part.

Thanks for your help.

**Comment by jems1140** [comment]
Hello,

Same problem, NvidiaTvShield with external Hdd, after the update all my files is not to found on Kodi, I try to delete refresh everything but it's like the same.
Did anybody know a quick solution ? 

Thanks

**Comment by fuzzard** [comment]
Try https://forum.kodi.tv/showthread.php?tid=371331&pid=3132727#pid3132727

**Comment by jems1140** [comment]
Thanks for you last comment, Yesterday I was searching hours.
But this help me now on my Nvidia Shield : 
On Android TV go to Settings -> Apps - Kodi -> Permissions -> Files and Media and select Allow all the time.

**Comment by jems1140** [comment]
Last update I had for example one movie all my screen with the pic of the movie and also the resume, know it's dissapeared...

**Comment by Raptor039** [comment]
> Try https://forum.kodi.tv/showthread.php?tid=371331&pid=3132727#pid3132727

Hi @fuzzard,

It's not the same problem I have. If it was, it would work with a fresh kodi 20 running on linux but that's not the case.

Nevertheless, permissions are ok on my Android TV, even if none of my media file are on a local drive. There are all located inside a nas accessible by kodi through smb.

**Comment by dirkdirk** [comment]
All I can tell you is what worked for me:  

Android TV > Settings > Device Preferences > Storage ... make sure "Scan for media automatically" is toggled on.

Immediately after toggling this on, Kodi could see all media files.

Hope this helps.

**Comment by thexai** [comment]
This is not general issue but a particular issue in "one" setup.

It should be closed and discussed in the forum as "support"

**Comment by trentmitchell1976** [comment]
Having identical issue after upgrade ... any solution?

**Comment by snaxilian** [comment]
I have this issue to. Clients Android TV and also on Windows (non UWP). Media files on a smb share and library information on a mariadb sql server.
The mentioned "solution" from @dirkdirk does not apply because media files are not local and the mentioned way to reach the settings are not available on my Android TV (Philips 55pus8601, latest available patches installed).

**Comment by Dominik-1980** [comment]
Same issue here on Shield TV

**Comment by Egregius** [comment]
Same issue here on Shield and Mac OS 13.1. 
Library empty and won't scan after setting the content and scraper.

**Comment by Raptor039** [comment]
@thexai maybe you could reopen this bug report and help us to investigate what's happen as it seems it's not anymore a "particular issue in one setup" as you can see with other comments?

Thanks for your help.

**Comment by Egregius** [comment]
Mine is working again!
Was going thru the kodi log on my Mac and found errors that the new database couldn't be created. 
I created a temporary new user with full admin rights on the MySQL server and put those credentials to the advancedsettings.xml.
After launching Kodi a new database with version 121 was made with all my data in it. 
Then changed the rights of my regular kody MySQL user to have full control on the new database and everything was back.

**Comment by Raptor039** [comment]
@Egregius glad to sea it works for you. In my case, the right are OK :
`MariaDB [(none)]> show grants for kodi;
+--------------------------------------------------------------------------------------------------------------+
| Grants for kodi@%                                                                                            |
+--------------------------------------------------------------------------------------------------------------+
| GRANT ALL PRIVILEGES ON *.* TO `kodi`@`%` IDENTIFIED BY PASSWORD '***************************************' |
+--------------------------------------------------------------------------------------------------------------+
1 row in set (0.001 sec)`

However, not all the tables were migrated from MyVideos119 to MyVideos121 and I don't understand why.

I will try to dissect debug log while migrating from a windows installation of Kodi 20 et report here if I can found something helping.

**Comment by fuzzard** [comment]
maybe https://forum.kodi.tv/showthread.php?tid=371463&pid=3134303#pid3134303 might be of relevance if the sql server itself has been updated at some point in the past.

**Comment by Raptor039** [comment]
@fuzzard it's ok for me 
```
$ sudo mysql_upgrade
This installation of MariaDB is already upgraded to 10.5.15-MariaDB.
There is no need to run mysql_upgrade again for 10.5.15-MariaDB.
You can use --force if you still want to run mysql_upgrade
```

**Comment by thexai** [comment]
Github if for report and track _confirmed_ bugs in Kodi code. 

Forum is for support, chat, etc.

Talk about users permissions in MariaDB server not seems a confirmed kodi bug.

**Comment by the-black-eagle** [comment]
@Raptor039 - Please get a proper debug log of the upgrade attempt and open a thread in the forum linking to the log.  Thanks!

**Comment by Raptor039** [comment]
@the-black-eagle Can you show me please the best way to get a debug log during the upgrade process?

Do I have to enable debug in kodi 19, attempt to upgrade and after what I'll have the proper log?

**Comment by the-black-eagle** [comment]
Yes, exactly that.  But, PLEASE, open a thread on the forum to continue this. Every message on here will be pinging over 100 kodi team members each time someone responds.  Therefore we would much prefer to support you on the forum (https://forum.kodi.tv/).  Thank you.

**Comment by trentmitchell1976** [comment]
When I checked my docker container for mariadb it had a message about running mysql_upgrade.  I opened a terminal window and ran the update then dropped the table and reopened kodi.  It updated and created the tables correctly this time.


Thanks for the help.

**Comment by boskabouter007** [comment]
I had the same problem. On Raspberry Pi 4 LibreElec upgraded from 10 tot 11.0.1 (Kodi Nexus v20.1)

I also checked the old db (MyVideos119) and the new created by the upgrade (MyVideos121).

- MariaDB [(none)]> show tables from MyVideos119;
- MariaDB [(none)]> show tables from MyVideos121;
The new db has only 30 tables while the old one has 37 tables. I don't know if it's ok because the schemas have evolved or if it's something else.

I've ran "mysql-upgrade" with did some table updates/fixing
After a reboot, all my Libraries reappeared.

Thanks for the valuable input.
